kratos/utilities/math_utils.h

```diff
@@ -1574,45 +1574,150 @@ class MathUtils
      }
 
     /**
-     * @brief Calculates the eigenvectors and eigenvalues of given symmetric TDimxTDim matrix
-     * @details The eigenvectors and eigenvalues are calculated using the iterative Gauss-Seidel-method
+     * @brief Calculates the product operation B'DB
+     * @param rA The resulting matrix
+     * @param rD The "center" matrix
+     * @param rB The matrices to be transposed
+     * @tparam TMatrixType1 The type of matrix considered (1)
+     * @tparam TMatrixType2 The type of matrix considered (2)
+     * @tparam TMatrixType3 The type of matrix considered (3)
+     */
+    template<class TMatrixType1, class TMatrixType2, class TMatrixType3>
+    static inline void BtDBProductOperation(
+        TMatrixType1& rA,
+        const TMatrixType2& rD,
+        const TMatrixType3& rB
+        )
+    {
+        // The sizes
+        const SizeType size1 = rB.size2();
+        const SizeType size2 = rB.size2();
+
+#ifdef KRATOS_USE_AMATRIX   // This macro definition is for the migration period and to be removed afterward please do not use it
+        KRATOS_WARNING_IF("BtDBProductOperation", rA.size1() != size1 || rA.size2() != size2) << "BtDBProductOperation has detected an incorrect size of your resulting matrix matrix. Please resize before compute" << std::endl;
```

Comment:
why different behaviour with AMAtrix?

Reply:
Because right now the AMatrix does not have a resize for bounded matrix. @philbucher proposed a dummy resize that throws an error. Waiting for @pooyan-dadvand